drivers/i2c/Kconfig.stm32

```diff
@@ -7,6 +7,7 @@ DT_COMPAT_ST_STM32_I2C_V2 := st,stm32-i2c-v2
 menuconfig I2C_STM32
 	bool "STM32 I2C driver"
 	default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_I2C_V1)) || $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_I2C_V2))
+	depends on SOC_FAMILY_STM32
```

Comment:
We generally set these in `soc/arm/st_stm32/common/Kconfig.defconfig.series`

Reply:
Do we?  Most of the other I2C drivers do it here.

Reply:
This is a depends not a default.  W/o a depends of some for its possible to configure the I2C_STM32 for example to be built on a NRF or NXP board / soc platform which obviously doesn't make any sense.

Reply:
ok, yes, mixed up things here.